backend/satisfactionCalculator.js

```diff
@@ -0,0 +1,77 @@
+const votes = require('./voteStore')
+const voteStorage = new votes()
+const alerter = require('./alerting.js')
+const alert = new alerter()
+const timebucket = require('timebucket')
+
+class calculator {
+
+    /**
+     * 
+     * @param {areaList[]} areaList 
+     * @param {votes[]} voteStore
+     * @abstract The basic idea is that it takes in the list of areas and all the current votes
+     *           then it runs through the areas list aggreates all the votes from that area and 
+     *           checks to see if the number of votes in the past hour passes the critical mass 
+     *           which for now is just half the area. 
+     *              
+     *           I would really liked this looked over as I think this is what's needed but 
+     *           I'm not fully sure.
+     */
+    areaCheck (areaList, voteStore) {
+        for(let j = 0; j < areaList.length; j++) {
+            let tempArray = new Array
+            for(let i = 0; i < voteStore.keys.length; i++) {
+                if (voteStore.keys[i].area == areaList[j]) {
+                    tempArray[i] = voteStore.keys[i]
+                }
+            }
+            let critMass = 0;
+            for(let i = 0; i < tempArray.length; i++) {
+                if (tempArray[i].timeStamp == timebucket('h').subtract(1) + '') {
+                    critMass++
+                }
+            }
+            if(areaList[i].size/2 == critMass) {
```

Comment:
What does `i` represent here?

Reply:
Also, isn't `areaList` an array of the strings of the area names? Wouldn't `areaList[i].size` return the characters in that particular area name?

Reply:
You're right it should be areaList[j] and the .size is referring to the number of people within the area. Just a stopover for the critical mass when half of the room votes within the last hour.